client/app/visualizations/chart/index.js

```diff
@@ -306,7 +306,7 @@ export default function init(ngModule) {
       sortX: true,
       legend: { enabled: true },
       yAxis: [{ type: 'linear' }, { type: 'linear', opposite: true }],
-      xAxis: { type: 'datetime', labels: { enabled: true } },
+      xAxis: { type: '-', labels: { enabled: true } },
```

Comment:
You need to add this option to the `xAxisScales` array (and probably change it to have labels for each one, because currently [we use the value itself](https://github.com/getredash/redash/blob/master/client/app/visualizations/chart/chart-editor.html#L164), which won't work for `-`).

Reply:
Okay that makes sense, @emtwo could take another look at this?